components/sup/src/manager/service/mod.rs

```diff
@@ -174,7 +174,9 @@ impl Service {
         manager_fs_cfg: Arc<manager::FsCfg>,
         organization: Option<&str>,
     ) -> Result<Service> {
-        let package = util::pkg::install_from_spec(&mut UI::default(), &spec)?;
+        // The package for a spec should already be installed.
```

Comment:
It's true to the extent that `hab svc start` and `hab svc load` are how you get a spec onto disk. If you manually create your own spec file, then it won't necessarily be installed, but then you've broken the rules and gone mucking around with the supervisor's internal state.

I suppose you could also get into a weird state if you've blown away directories under `/hab/pkgs`, too...

The old way would have potentially blocked the main thread, for what it's worth. Trying to install here incurs a request to Builder each time, because it'll need to find what the latest version of the package is. Then that also starts pulling in update logic into installation logic (because you don't want to update all the time unless you're an "at-once" service), and generally makes things more complex.

Maybe we could bubble up an explanatory error here if the call to `PackageInstall::load` returns an error?

Reply:
That's right about the previous behavior blocking the main thread. We had a different set of assumptions at that point in time (threading was going to happen until we stopped short).

Good explanation, I'm sold.